### PR TITLE
[4.3] Fix missing filter for subject

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #41718 .

### Summary of Changes
Add missing filter for subject into consents model

### Testing Instructions
see #41718

### Actual result BEFORE applying this Pull Request
Filtering for subject not working

### Expected result AFTER applying this Pull Request
Filtering for subject works. 

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_privacy/src/Model/ConsentsModel.php [hunk 1]
```diff
@@ -105,6 +105,13 @@ protected function getListQuery()
                 ->bind(':state', $state, ParameterType::INTEGER);
         }
 
+        $subject = $this->getState('filter.subject');
+
+        if (!empty($subject)) {
+            $query->where($db->quoteName('a.subject') . ' = :subject')
+                ->bind(':subject', $subject, ParameterType::STRING);
+        }
+
```
Comment (richard67) [inline administrator/components/com_privacy/src/Model/ConsentsModel.php]:
```suggestion
        }

```
Fix PHPCS no space in empty line.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on c64bb0600fce979802abdd1f1bb095ab44bb04b1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41726">issues.joomla.org/tracker/joomla-cms/41726</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on c64bb0600fce979802abdd1f1bb095ab44bb04b1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41726">issues.joomla.org/tracker/joomla-cms/41726</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41726">issues.joomla.org/tracker/joomla-cms/41726</a>.</sub>

**Comment by obuisard** [comment]
Thank you Christiane @chmst for the PR!

**Pull request merged**